lib/ruby_smb/smb1/file.rb

```diff
@@ -0,0 +1,193 @@
+      # @!attribute [rw] last_write
+      #   @return [DateTime]
+      attr_accessor :last_write
+
+      # The actual size, in bytes, of the file
+      # @!attribute [rw] size
+      #   @return [Integer]
+      attr_accessor :size
+
+      # The size in bytes that the file occupies on disk
+      # @!attribute [rw] size_on_disk
+      #   @return [Integer]
+      attr_accessor :size_on_disk
+
+      def initialize(tree:, response:, name:)
+        raise ArgumentError, 'No tree provided' if tree.nil?
+        raise ArgumentError, 'No response provided' if response.nil?
+        raise ArgumentError, 'No file name provided' if name.nil?
+
+        @tree = tree
+        @name = name
+
+        @attributes   = response.parameter_block.ext_file_attributes
+        @fid          = response.parameter_block.fid
+        @last_access  = response.parameter_block.last_access_time.to_datetime
+        @last_change  = response.parameter_block.last_change_time.to_datetime
+        @last_write   = response.parameter_block.last_write_time.to_datetime
+        @size         = response.parameter_block.end_of_file
+        @size_on_disk = response.parameter_block.allocation_size
+      end
+
+      # Appends the supplied data to the end of the file.
+      #
+      # @param data [String] the data to write to the file
+      # @return [WindowsError::ErrorCode] the NTStatus code returned from the operation
+      def append(data:)
+        write(data: data, offset: @size)
+      end
+
+      # Closes the handle to the remote file.
+      #
+      # @return [WindowsError::ErrorCode] the NTStatus code returned by the operation
+      def close
+        close_request = set_header_fields(RubySMB::SMB1::Packet::CloseRequest.new)
+        raw_response  = @tree.client.send_recv(close_request)
+        response = @tree.client.parse_response(response_packet: RubySMB::SMB1::Packet::CloseResponse, raw_response: raw_response)
+        response.status_code
+      end
+
+      # Read from the file, a specific number of bytes
+      # from a specific offset. If no parameters are given
+      # it will read the entire file.
+      #
+      # @param bytes [Integer] the number of bytes to read
+      # @param offset [Integer] the byte offset in the file to start reading from
+      # @return [String] the data read from the file
+      def read(bytes: @size, offset: 0)
+        atomic_read_size = if bytes > @tree.client.max_buffer_size
+                             @tree.client.max_buffer_size
+                           else
+                             bytes
+                           end
+        remaining_bytes = bytes
+        data = ''
+
+        loop do
+          read_request = read_packet(read_length: atomic_read_size, offset: offset)
+          raw_response = @tree.client.send_recv(read_request)
+          response = @tree.client.parse_response(response_packet: RubySMB::SMB1::Packet::ReadAndxResponse, raw_response: raw_response)
```

Comment:
In smb2 implementation we use the Packet class to read the response. https://github.com/rapid7/ruby_smb/blob/master/lib/ruby_smb/smb2/file.rb#L105

Reply:
I removed #parse_response helper, used the Packet class to read the response and added the validation logic. Thanks! Addressed in 4bb00fb.